swap/swap_test.go

```diff
@@ -305,7 +314,7 @@ func newTestCheque() *Cheque {
 
 func TestEncodeCheque(t *testing.T) {
 	// setup test swap object
-	swap, dir := createTestSwap(t)
+	swap, dir := createTestSwap(t, nil)
```

Comment:
If you add the two functions as described above, then there would be no reason to call `createTestSwap` with `nil` parameters which are not very readable.